app/models/check.rb

```diff
@@ -130,36 +87,21 @@ def push_check_metrics
     Metrics.count("scan.check.count", 1, metric_tags)
   end
 
-  def verify_state_machine
-    if status
-      db_check = Check.find(id)
-      # There is a status change. Need to check state machine
-      if db_check[:status] != self[:status]
-        case self[:status]
-        when "QUEUED"
-          db_check.enqueue
-        when "ASSIGNED"
-          db_check.assign
-        when "MALFORMED"
-          db_check.malform
-        when "RUNNING"
-          db_check.run
-        when "TIMEOUT"
-          db_check.timeout
-        when "FAILED"
-          db_check.fail
-        when "ABORTED"
-          db_check.abort
-        when "PURGING"
-          db_check.purge
-        when "KILLED"
-          db_check.kill
-        when "FINISHED"
-          db_check.finish
-        when "INCONCLUSIVE"
-          db_check.inconclusive
-        end
-      end
-    end
+  def verify_transition
+    return true unless status
+    db_check = Check.find(id)
+    # We do not allow status changes if the check is already finished.
+    return false if @@terminal_statuses.include?(db_check[:status].intern)
+    # Check if the status change is allowed.
+    # It will be only if the new estatus is "greater" than the current one.
+    return true if db_check[:status] == self[:status]
+    return true if @@terminal_statuses.include?(self[:status].intern)
+
+    allowed = @@allowed_trasitions[db_check[:status].intern]
+
+    return false unless allowed
+    return false unless allowed.include?(self[:status].intern)
```

Comment:
It is true but, theoretically if the self[:status] is ABORTED, that condition will never be evaluated, because
one of those conditions, that are above this one, must be true:
 return false if @@terminal_statuses.include?(db_check[:status].intern)
or 
return true if @@terminal_statuses.include?(self[:status].intern)

is that right?